src/AspNet.Security.OAuth.SuperOffice/SuperOfficeAuthenticationHandler.cs

```diff
@@ -0,0 +1,159 @@
+﻿/*
+ * Licensed under the Apache License, Version 2.0 (http://www.apache.org/licenses/LICENSE-2.0)
+ * See https://github.com/aspnet-contrib/AspNet.Security.OAuth.Providers
+ * for more information concerning the license and the contributors participating to this project.
+ */
+
+using System;
+using System.Collections.Generic;
+using System.Globalization;
+using System.IdentityModel.Tokens.Jwt;
+using System.Linq;
+using System.Net.Http;
+using System.Net.Http.Headers;
+using System.Security.Claims;
+using System.Text.Encodings.Web;
+using System.Text.Json;
+using System.Threading.Tasks;
+using JetBrains.Annotations;
+using Microsoft.AspNetCore.Authentication;
+using Microsoft.AspNetCore.Authentication.OAuth;
+using Microsoft.Extensions.Logging;
+using Microsoft.Extensions.Options;
+using Microsoft.IdentityModel.Tokens;
+
+namespace AspNet.Security.OAuth.SuperOffice
+{
+    /// <summary>
+    /// Defines a handler for authentication using SuperOffice.
+    /// </summary>
+    public class SuperOfficeAuthenticationHandler : OAuthHandler<SuperOfficeAuthenticationOptions>
+    {
+        private readonly JwtSecurityTokenHandler _tokenHandler;
```

Comment:
As I said in a previous comment, this shouldn't be injected as a constructor parameter but stored as an option in `SuperOfficeAuthenticationOptions`. Otherwise, users won't be able to use a customized version for a specific authentication handler.

Reply:
Shouldn't implies an opinionated suggestion. Can you refer me to any documentation that provides alternative guidance in this case scenario?

Reply:
OK. I have moved this into SuperOfficeAuthenticationOptions. If none is provider, the default JwtSecurityTokenHandler is instantiated in the IPostConfigureOptions implementation.